File: js/BST_print_level_order_one_line.js

class Node {
    constructor(val, left = null, right = null) {
        this.val = val
        this.left = left
        this.right = right
    }
}

function print_level_order (root) {
    let levelOrder = ''
    const queue = [root]

    while (queue.length) {
        const node = queue.pop()

        if (node.left) {
            queue.unshift(node.left)
        }

        if (node.right) {
            queue.unshift(node.right)
        }

        levelOrder += node.val
    }

    return levelOrder
}

const rootNode = new Node(1, new Node(2), new Node(3, new Node(4), new Node(5)))
const res = print_level_order(rootNode)

console.log('output:', res)

File: py/BST_print_level_order_one_line.py
from collections import deque

class Node(object):
  def __init__(self, val):
    self.val = val
    self.left = None
    self.right = None

  def __str__(self):
    q = deque([self])
    str = ''

    while len(q):
      node = q.pop()

      if node.left:
        q.appendleft(node.left)

      if node.right:
        q.appendleft(node.right)

      str += node.val

    return str


tree = Node('a')
tree.left = Node('b')
tree.right = Node('c')
tree.left.left = Node('d')
tree.left.right = Node('e')
tree.right.left = Node('f')
tree.right.right = Node('g')

print(tree)